lib/Twig/Node/Expression/Call.php

```diff
@@ -17,34 +17,47 @@ protected function compileCallable(Twig_Compiler $compiler)
         $callable = $this->getAttribute('callable');
 
         $closingParenthesis = false;
-        if (is_string($callable) && false === strpos($callable, '::')) {
-            $compiler->raw($callable);
-        } else {
-            list($r, $callable) = $this->reflectCallable($callable);
-            if ($r instanceof ReflectionMethod && is_string($callable[0])) {
-                if ($r->isStatic()) {
-                    $compiler->raw(sprintf('%s::%s', $callable[0], $callable[1]));
-                } else {
-                    $compiler->raw(sprintf('$this->env->getRuntime(\'%s\')->%s', $callable[0], $callable[1]));
-                }
-            } elseif ($r instanceof ReflectionMethod && $callable[0] instanceof Twig_ExtensionInterface) {
-                // For BC/FC with namespaced aliases
-                $class = (new ReflectionClass(get_class($callable[0])))->name;
-                if (!$compiler->getEnvironment()->hasExtension($class)) {
-                    throw new Twig_Error_Runtime(sprintf('The "%s" extension is not enabled.', $class));
-                }
+        $compileInline = false;
+
+        if ($this->hasAttribute('inline') && $this->getAttribute('inline')) {
+            try {
+                $this->compileInline($compiler, $callable);
+                $compileInline = true;
+            } catch (RuntimeException $e) {
+                // Continue standard source compilation
+            }
+        }
 
-                $compiler->raw(sprintf('$this->extensions[\'%s\']->%s', ltrim($class, '\\'), $callable[1]));
+        if (!$compileInline) {
```

Comment:
instead of indenting all this logic in a if block to cancel all the normal compilation when it was compiled inline, you should rather return earlier after compiling inline.